comms/src/builder/builder.rs

```diff
@@ -0,0 +1,491 @@
+        Ok(Arc::new(peer_manager))
+    }
+
+    fn make_control_service(
+        &mut self,
+        node_identity: Arc<NodeIdentity<CommsPublicKey>>,
+    ) -> Option<ControlService<MType>>
+    {
+        self.control_service_config_factory
+            .take()
+            .map(|f| f.make())
+            .map(|config| ControlService::new(self.comms_context.clone(), node_identity, config))
+    }
+
+    fn make_connection_manager(
+        &mut self,
+        node_identity: Arc<NodeIdentity<CommsPublicKey>>,
+        peer_manager: Arc<PeerManager<CommsPublicKey, CommsDataStore>>,
+        config: PeerConnectionConfig,
+    ) -> Arc<ConnectionManager>
+    {
+        Arc::new(ConnectionManager::new(
+            self.comms_context.clone(),
+            node_identity,
+            peer_manager,
+            config,
+        ))
+    }
+
+    fn make_peer_connection_config(&mut self) -> PeerConnectionConfig {
+        let mut config = self
+            .peer_conn_config_factory
+            .take()
+            .map(|f| f.make())
+            .unwrap_or_default();
+        // If the message_sink_address is not set (is default) set it to a random inproc address
+        if config.message_sink_address.is_default() {
+            config.message_sink_address = InprocAddress::random();
+        }
+        config
+    }
+
+    fn make_node_identity(&mut self) -> Result<Arc<NodeIdentity<CommsPublicKey>>, CommsBuilderError> {
+        self.node_identity
+            .take()
+            .map(Arc::new)
+            .ok_or(CommsBuilderError::NodeIdentityNotSet)
+    }
+
+    fn make_oms(
+        &self,
+        node_identity: Arc<NodeIdentity<CommsPublicKey>>,
+        message_sink_address: InprocAddress,
+        peer_manager: Arc<PeerManager<CommsPublicKey, CommsDataStore>>,
+    ) -> Result<OutboundMessageService, CommsBuilderError>
+    {
+        OutboundMessageService::new(
+            self.comms_context.clone(),
+            node_identity,
+            message_sink_address,
+            peer_manager,
+        )
+        .map_err(CommsBuilderError::OutboundMessageServiceError)
+    }
+
+    fn make_omp(
+        &mut self,
+        message_sink_address: InprocAddress,
+        peer_manager: Arc<PeerManager<CommsPublicKey, CommsDataStore>>,
+        connection_manager: Arc<ConnectionManager>,
+    ) -> Result<OutboundMessagePool, CommsBuilderError>
+    {
+        let config = self.omp_config_factory.take().map(|f| f.make()).unwrap_or_default();
+
+        OutboundMessagePool::new(
+            config,
+            self.comms_context.clone(),
+            // OMP can requeue back onto itself
+            message_sink_address.clone(),
+            message_sink_address.clone(),
+            peer_manager,
+            connection_manager,
+        )
+        .map_err(CommsBuilderError::OutboundMessagePoolError)
+    }
+
+    pub fn make_inbound_message_service(
```

Comment:
Nit: I prefer consistency, you were using acronyms before so rather use IMS here

Reply:
Done